packages/components/src/internal/components/search/FilterFacetedSelector.tsx

```diff
@@ -11,36 +11,49 @@ import { ComponentsAPIWrapper, getDefaultAPIWrapper } from '../../APIWrapper';
 
 import { ALL_VALUE_DISPLAY, EMPTY_VALUE_DISPLAY, getCheckedFilterValues, getUpdatedChooseValuesFilter } from './utils';
 
+const MAX_DISTINCT_FILTER_OPTIONS = 250;
+
 interface Props {
     api?: ComponentsAPIWrapper;
     fieldFilters: Filter.IFilter[];
     fieldKey: string;
+    canBeBlank: boolean;
     onFieldFilterUpdate?: (newFilters: Filter.IFilter[], index) => void;
     selectDistinctOptions: Query.SelectDistinctOptions;
     showSearchLength?: number; // show search box if number of unique values > N
 }
 
 export const FilterFacetedSelector: FC<Props> = memo(props => {
-    const { api, selectDistinctOptions, fieldKey, fieldFilters, onFieldFilterUpdate, showSearchLength } = props;
+    const { api, canBeBlank, selectDistinctOptions, fieldKey, fieldFilters, onFieldFilterUpdate, showSearchLength } = props;
 
     const [fieldDistinctValues, setFieldDistinctValues] = useState<string[]>(undefined);
     const [error, setError] = useState<string>(undefined);
     const [searchStr, setSearchStr] = useState<string>(undefined);
+    const [allShown, setAllShown] = useState<boolean>(true);
 
     useEffect(() => {
+        setDistinctValues(true);
+    }, [fieldKey]); // on fieldKey change, reload selection values
+
+    const setDistinctValues = useCallback((checkAllShown: boolean, searchStr?: string) => {
+        const filterArray = searchStr ? [Filter.create(fieldKey, searchStr, Filter.Types.CONTAINS)].concat(selectDistinctOptions?.filterArray) : fieldFilters;
```

Comment:
If searchStr is blank, should also use selectDistinctOptions?.filterArray, not fieldFilters

Reply:
oops. Fixed.